edi_exchange_template_oca/tests/test_backend_and_type.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2025 Camptocamp SA
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import SavepointCase
+from odoo.tools import mute_logger
+
+
+class TestExchangeType(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.backend_type = cls.env.ref("edi_oca.demo_edi_backend_type")
+        cls.backend = cls.env.ref("edi_oca.demo_edi_backend")
+        cls.type_out1 = cls.env["edi.exchange.type"].create(
+            {
+                "name": "Type output 1",
+                "direction": "output",
+                "code": "test_type_out1",
+                "exchange_file_ext": "txt",
+                "backend_type_id": cls.backend_type.id,
+            }
+        )
+        cls.type_out2 = cls.env["edi.exchange.type"].create(
+            {
+                "name": "Type output 2",
+                "direction": "output",
+                "code": "test_type_out2",
+                "exchange_file_ext": "txt",
+                "backend_type_id": cls.backend_type.id,
+            }
+        )
+        model = cls.env["edi.exchange.template.output"]
+        qweb_tmpl = cls.env["ir.ui.view"].create(
+            {
+                "type": "qweb",
+                "key": "edi_exchange.test_output1",
+                "arch": """
+            <t t-name="edi_exchange.test_output1">
+                TEST
+            </t>
+            """,
+            }
+        )
+        cls.tmpl_out1 = model.create(
+            {
+                "code": "tmpl_test_type_out1",
+                "name": "Out 1",
+                "backend_type_id": cls.backend_type.id,
+                "template_id": qweb_tmpl.id,
+                "output_type": "txt",
+            }
+        )
+        cls.tmpl_out2 = model.create(
+            {
+                "code": "tmpl_test_type_out2",
+                "name": "Out 2",
+                "backend_type_id": cls.env.ref("edi_oca.demo_edi_backend_type").id,
+                "template_id": qweb_tmpl.id,
+                "output_type": "txt",
+            }
+        )
+        vals = {
+            # doesn't matter what model we use
+            "model": cls.env.user._name,
+            "res_id": cls.env.user.id,
+            "type_id": cls.type_out2.id,
+        }
+        cls.record1 = cls.backend.create_record("test_type_out1", vals)
+        vals = {
+            # doesn't matter what model we use
+            "model": cls.env.user._name,
+            "res_id": cls.env.user.id,
+            "type_id": cls.type_out2.id,
+        }
+        cls.record2 = cls.backend.create_record("test_type_out2", vals)
+
+    # TODO: getting a template via code or relying on a fallback is deprecated
```

Comment:
Any test regarding testing that a msg is logged when using code/fallback?

Reply:
I'm testing the fallback itself... and yeah ok, but I want to trash this code... so I won't spend time now.